Core dump when trying to share the screen

I'm getting a reproducible core dump when trying to share my screen via WebRTC with the following error message:
```
2022/02/15 11:27:02 [ERROR] - xdg-desktop-portal-wlr: failed to convert wl_shm format 0x34324742 to spa_video_format
[1]    19548 abort (core dumped)  /usr/lib/xdg-desktop-portal-wlr -r
```
Currently, I'm using manjaro with the -git version of the following packages:
```
sway-git - r6879.f707f583-1
wlroots-git - 0.16.0.r5365.252b2348-1
xdg-desktop-portal-wlr-git - v0.5.0.r24.gc34d098-1
```
All other packages are the most up to date normal versions.

I tried to do this
```diff
diff --git a/src/screencast/screencast_common.c b/src/screencast/screencast_common.c
index 038cf7a..426ca69 100644
--- a/src/screencast/screencast_common.c
+++ b/src/screencast/screencast_common.c
@@ -302,9 +302,9 @@ uint32_t xdpw_format_drm_fourcc_from_wl_shm(enum wl_shm_format format) {
    case WL_SHM_FORMAT_BGRA1010102:
        return (uint32_t)format;
    default:
-       logprint(ERROR, "xdg-desktop-portal-wlr: unsupported wl_shm "
+       logprint(WARN, "xdg-desktop-portal-wlr: unsupported wl_shm "
            "format 0x%08x", format);
-       abort();
+       return (uint32_t)format;
    }
 }
```
but then xdpw cannot convert the format to `spa_video_format`.
```
2022/04/28 21:41:14 [TRACE] - wlroots: callbacks registered
2022/04/28 21:41:14 [TRACE] - wlroots: buffer event handler
2022/04/28 21:41:14 [WARN] - xdg-desktop-portal-wlr: unsupported wl_shm format 0x34324742
2022/04/28 21:41:14 [TRACE] - wlroots: linux_dmabuf event handler
2022/04/28 21:41:14 [TRACE] - wlroots: buffer_done event handler
2022/04/28 21:41:14 [TRACE] - wlroots: finish screencopy
2022/04/28 21:41:14 [TRACE] - wlroots: frame destroyed
2022/04/28 21:41:14 [INFO] - wlroots: num_modififiers 13
2022/04/28 21:41:14 [ERROR] - xdg-desktop-portal-wlr: failed to convert drm format 0x34324742 to spa_video_format
<SIGABRT>
```
I'm not sure how it should be converted [considering this comment](https://github.com/emersion/xdg-desktop-portal-wlr/issues/190#issuecomment-1073565870)? So is it, in fact, possible to somehow use this video format in `libwebrtc`, considering comments that claims it works in KDE and GNOME? Sorry for being a little lost here.

--- a/src/screencast/screencast_common.c
+++ b/src/screencast/screencast_common.c
@@ -279,6 +279,8 @@ enum wl_shm_format xdpw_format_wl_shm_from_drm_fourcc(uint32_t format) {
 	case DRM_FORMAT_ABGR2101010:
 	case DRM_FORMAT_RGBA1010102:
 	case DRM_FORMAT_BGRA1010102:
+	case DRM_FORMAT_BGR888:
+	case DRM_FORMAT_RGB888:
 		return (enum wl_shm_format)format;
 	default:
 		logprint(ERROR, "xdg-desktop-portal-wlr: unsupported drm "
@@ -308,6 +310,8 @@ uint32_t xdpw_format_drm_fourcc_from_wl_shm(enum wl_shm_format format) {
 	case WL_SHM_FORMAT_ABGR2101010:
 	case WL_SHM_FORMAT_RGBA1010102:
 	case WL_SHM_FORMAT_BGRA1010102:
+	case WL_SHM_FORMAT_BGR888:
+	case WL_SHM_FORMAT_RGB888:
 		return (uint32_t)format;
 	default:
 		logprint(ERROR, "xdg-desktop-portal-wlr: unsupported wl_shm "
@@ -352,6 +356,10 @@ enum spa_video_format xdpw_format_pw_from_drm_fourcc(uint32_t format) {
 		return SPA_VIDEO_FORMAT_RGBA_102LE;
 	case DRM_FORMAT_BGRA1010102:
 		return SPA_VIDEO_FORMAT_BGRA_102LE;
+	case DRM_FORMAT_BGR888:
+		return SPA_VIDEO_FORMAT_RGB;
+	case DRM_FORMAT_RGB888:
+		return SPA_VIDEO_FORMAT_BGR;
 	default:
 		logprint(ERROR, "xdg-desktop-portal-wlr: failed to convert drm "
 			"format 0x%08x to spa_video_format", format);

